### DISCUSSION TITLE
Proposal: Introducing HasDDDFactory Trait for Factory Creation

### DISCUSSION THREAD

**Discussion by meditto** [lunarstorm/laravel-ddd#30] [Ideas]
Hello, I hope you're all doing well. I've been using `laravel-ddd` for the past few days and have found it to be a valuable tool for my Laravel projects. I'd like to suggest a feature enhancement that I believe could further improve the package's usability.
The current process involves generating the `BaseMdel`. I believe this process could be simplified and made more user-friendly by introducing a new trait, `HasDDDFactory`.

The proposed `HasDDDFactory` trait would serve as a drop-in replacement for the existing `HasFactory` trait. It would automatically handle the detection of the appropriate factory path just like the `BaseModel`.

The flexibility of the `HasDDDFactory` trait could also extend to models that don't directly extend the Model object, such as the User model. This adaptability adds to its potential usefulness.

![image](https://github.com/lunarstorm/laravel-ddd/assets/26086016/136c4da9-e666-42e4-bf41-36f4935819a4)

I would greatly appreciate your thoughts and insights on this proposal.

**Comment by pelmered** [comment]
I like this.

**Comment by jaspertey** [comment]
This is now possible in v0.10.0 using `Lunarstorm\LaravelDDD\Factories\HasDomainFactory`: https://github.com/lunarstorm/laravel-ddd/releases/tag/v0.10.0
